test/quantization/fx/test_equalize_fx.py

```diff
@@ -267,6 +270,43 @@ def test_input_weight_equalization_prepare(self):
                  (FunctionalConvReluConvModel, two_F_layer_node_occurrence)]
 
         for (M, node_occurrence) in tests:
+            m = M().eval()
+            prepared = prepare_fx(m, specific_qconfig_dict, equalization_qconfig_dict=default_equalization_qconfig_dict)
+            self.checkGraphModuleNodes(prepared, expected_node_occurrence=node_occurrence)
+
+    def test_input_weight_equalization_branching(self):
+        """ Tests that graphs containing branches are prepared correctly.
+        Specifically, equalization observers should not be inserted in front of
+        branches in which both initial layers in the branches plan to be
+        quantized.
+        """
+
+        class AddBranchingModel(nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.linear1 = nn.Linear(5, 5)
+
+            def forward(self, x):
+                y = self.linear1(x)
+                z = torch.add(x, 5)
+                return torch.add(y, z)
+
+        multiple_ops_node_occurrence = {
+            ns.call_module(_InputEqualizationObserver): 2,
+            ns.call_module(MinMaxObserver): 7,
+        }
+
+        add_branching_node_occurrence = {
+            ns.call_module(_InputEqualizationObserver): 1,
+            ns.call_module(MinMaxObserver): 2,
+        }
+
+        tests = [
+            (ModelMultipleOpsNoAvgPool, default_qconfig_dict, multiple_ops_node_occurrence),
+            (AddBranchingModel, specific_qconfig_dict, add_branching_node_occurrence),
```

Comment:
Trying to understand these tests a little better. Is ModelMultipleOpsNoAvgPool supposed to test the case where we don't equalize across branches (print warning) since all nodes are being quantized? And is AddBranchingModel supposed to test that we still equalize in a branch if the other branch is in FP32?
Maybe making the names a bit more descriptive and adding some comments might help someone trying to read this code.

Reply:
Yes, I will add print statements to ensure that these are better understood!